src/shell/interpreter.zig

```diff
@@ -10107,6 +10120,744 @@ pub const Interpreter = struct {
+            const EbusyState = struct {
+                tasks: std.ArrayListUnmanaged(*ShellCpTask) = .{},
+                absolute_targets: std.StringArrayHashMapUnmanaged(void) = .{},
+                absolute_srcs: std.StringArrayHashMapUnmanaged(void) = .{},
+
+                pub fn deinit(this: *EbusyState) void {
+                    this.tasks.deinit(bun.default_allocator);
+                    for (this.absolute_targets.keys()) |tgt| {
+                        bun.default_allocator.free(tgt);
+                    }
+                    this.absolute_targets.deinit(bun.default_allocator);
+                    for (this.absolute_srcs.keys()) |tgt| {
+                        bun.default_allocator.free(tgt);
+                    }
+                    this.absolute_srcs.deinit(bun.default_allocator);
+                }
+            };
+
+            pub fn start(this: *Cp) Maybe(void) {
+                const maybe_filepath_args = switch (this.opts.parse(this.bltn.argsSlice())) {
+                    .ok => |args| args,
+                    .err => |e| {
+                        const buf = switch (e) {
+                            .illegal_option => |opt_str| this.bltn.fmtErrorArena(.cp, "illegal option -- {s}\n", .{opt_str}),
+                            .show_usage => Builtin.Kind.cp.usageString(),
+                            .unsupported => |unsupported| this.bltn.fmtErrorArena(.cp, "unsupported option, please open a GitHub issue -- {s}\n", .{unsupported}),
+                        };
+
+                        _ = this.writeFailingError(buf, 1);
+                        return Maybe(void).success;
+                    },
+                };
+
+                if (maybe_filepath_args == null or maybe_filepath_args.?.len <= 1) {
+                    _ = this.writeFailingError(Builtin.Kind.cp.usageString(), 1);
+                    return Maybe(void).success;
+                }
+
+                const args = maybe_filepath_args orelse unreachable;
+                const paths_to_copy = args[0 .. args.len - 1];
+                const tgt_path = std.mem.span(args[args.len - 1]);
+
+                this.state = .{ .exec = .{
+                    .target_path = tgt_path,
+                    .paths_to_copy = paths_to_copy,
+                } };
+
+                this.next();
+
+                return Maybe(void).success;
+            }
+
+            pub fn ignoreEbusyErrorIfPossible(this: *Cp) void {
+                if (!bun.Environment.isWindows) @compileError("dont call this plz");
+
+                if (this.state.ebusy.idx < this.state.ebusy.state.tasks.items.len) {
+                    outer_loop: for (this.state.ebusy.state.tasks.items[this.state.ebusy.idx..], 0..) |task_, i| {
+                        const task: *ShellCpTask = task_;
+                        assert(task.tgt_absolute != null);
+                        const failure_src = task.src_absolute.?;
+                        const failure_tgt = task.tgt_absolute.?;
```

Comment:
why do we assert the target but not the source? do we need the assert at all if it's only going to be in debug builds anyway, which check `.?` ?